test/test_nn.py

```diff
@@ -3809,10 +3810,18 @@ def get_modules():
 
                 # TEST TRAINING BEHAVIOR
 
-                # run forward again and assert that u and v are updated
+                # We perform GD first to modify the initial matrix
+                wrapped_m(input).sum().backward()
+                with torch.no_grad():
+                    for p in wrapped_m.parameters():
+                        if p.requires_grad:
+                            p.add_(- p.grad, alpha=0.01)
```

Comment:
nit: extra space here?

Reply:
I actually went on an changed this code with calls to `SGD` as suggested by @albanD in https://github.com/pytorch/pytorch/pull/58488#discussion_r647458661